validator/src/journal/block_scheduler.rs

```diff
@@ -0,0 +1,260 @@
+/*
+ * Copyright 2018 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ------------------------------------------------------------------------------
+ */
+
+use std::collections::{HashMap, HashSet};
+use std::sync::RwLock;
+
+use block::Block;
+use journal::block_manager::BlockManager;
+use journal::block_wrapper::BlockStatus;
+use metrics;
+
+lazy_static! {
+    static ref COLLECTOR: metrics::MetricsCollectorHandle =
+        metrics::get_collector("sawtooth_validator.block_validator");
+}
+
+pub struct BlockScheduler<F: Fn(&str) -> BlockStatus> {
+    state: RwLock<BlockSchedulerState<F>>,
+}
+
+impl<F: Fn(&str) -> BlockStatus> BlockScheduler<F> {
+    pub fn new(block_manager: BlockManager) -> Self {
+        BlockScheduler {
+            state: RwLock::new(BlockSchedulerState {
```

Comment:
Does this struct need to be guarded by a lock, or does the owning structure (i.e. the future port of the Block validator) need to manage the a lock anyway?

Reply:
In the python version the BlockScheduler has its own lock.